dag-cbor/src/decode.rs

```diff
@@ -485,29 +468,25 @@ impl TryReadCbor for Ipld {
             // Major type 5: a map of pairs of data items
             0xa0..=0xbb => {
                 let len = read_len(r, major - 0xa0)?;
-                #[cfg(feature = "unleashed")]
-                if let Ok(map) = read_map(r, len as usize) {
-                    return Ok(Some(Self::StringMap(map)));
-                } else {
-                    // TODO this is a hack
-                    r.seek(SeekFrom::Current(-1))?;
-                    return Ok(Some(Self::IntegerMap(read_map(r, len as usize)?)));
+                if len > 0 {
+                    let pos = r.seek(SeekFrom::Current(0))?;
+                    #[cfg(feature = "unleashed")]
+                    if let Ok(map) = read_map(r, len as usize) {
+                        return Ok(Self::IntegerMap(map));
+                    }
+                    r.seek(SeekFrom::Start(pos))?;
```

Comment:
I think this whole if statement could be behind the `unleashed` feature. It is seeking to the current position if `unleashed` isn't enabled.